lib/platforms.nix

```diff
@@ -8,8 +8,9 @@ rec {
   openbsd = ["i686-openbsd" "x86_64-openbsd"];
   netbsd = ["i686-netbsd" "x86_64-netbsd"];
   cygwin = ["i686-cygwin" "x86_64-cygwin"];
-  unix = linux ++ darwin ++ freebsd ++ openbsd;
-  all = linux ++ darwin ++ cygwin ++ freebsd ++ openbsd;
+  illumos = ["x86_64-solaris"];
```

Comment:
Heh, I tried googling `x86_64-solaris`, `x86_64-sunos`, `x86_64-illumos`, and all do have references :-)

I think there's no use in trying to support the (proprietary) version. When someone needs it, (s)he can (try to) add it.